server/datastore/mysql/migrations/tables/20250410104321_UpdateMacOSSoftwareNames.go

```diff
@@ -0,0 +1,210 @@
+package tables
+
+import (
+	"database/sql"
+	"fmt"
+
+	"github.com/jmoiron/sqlx"
+	"github.com/jmoiron/sqlx/reflectx"
+	"github.com/pkg/errors"
+)
+
+func init() {
+	MigrationClient.AddMigration(Up_20250410104321, Down_20250410104321)
+}
+
+// TODO(JVE): add bundleid != null to all constraints
+func Up_20250410104321(tx *sql.Tx) error {
+	titleStmt := `UPDATE software_titles SET name = TRIM( TRAILING '.app' FROM name ) WHERE source = 'apps' AND bundle_identifier IS NOT NULL`
+	_, err := tx.Exec(titleStmt)
+	if err != nil {
+		return fmt.Errorf("updating software_titles.name: %w", err)
+	}
+
+	dupeIDsStmt := `SELECT
+					s1.id AS id, s1.bundle_identifier AS bundle_identifier FROM software s1
+					JOIN software s2 ON s1.bundle_identifier = s2.bundle_identifier
+						AND s1.version = s2.version
+						AND s1.title_id = s2.title_id
+						AND s1.source = s2.source
+						AND` + " s1.`release` = s2.`release` " +
+		`AND s1.arch = s2.arch
+						AND s1.vendor = s1.vendor
+						AND s1.browser = s2.browser
+						AND s1.extension_id = s2.extension_id
+				WHERE
+					s1.source = 'apps'
+				GROUP BY
+					id
+				HAVING
+					COUNT(*) > 1`
+
+	txx := sqlx.Tx{Tx: tx, Mapper: reflectx.NewMapperFunc("db", sqlx.NameMapper)}
```

Comment:
I'm a bit confused, why are you creating another transaction here? Don't the migrations automatically have a transactions `tx` already?
Also mysql doesn't allow nested transactions:
https://dev.mysql.com/doc/refman/8.4/en/implicit-commit.html
`
Transactions cannot be nested. This is a consequence of the implicit commit performed for any current transaction when you issue a [START TRANSACTION](https://dev.mysql.com/doc/refman/8.4/en/commit.html) statement or one of its synonyms.
`

Reply:
@ksykulev I think this is the pattern we use in migrations to do `SELECT`s, e.g. https://github.com/fleetdm/fleet/blob/ba6b6e12cc79cb1799b1304ce2dca04f27926408/server/datastore/mysql/migrations/tables/20221227163855_CleanupEmptyMobileDeviceManagementSolutions.go#L20
I think the `sqlx.Tx` is just a wrapper around a plain ol' `sql.Tx`, so it doesn't make a new transaction, just gives us the ability to do things like `txx.Select` or `txx.Get` without having to iterate over the raw rows